gk/bpf.c

```diff
@@ -156,7 +193,11 @@ gk_bpf_encapsulate(struct gk_bpf_pkt_ctx *ctx, int priority)
 	if (unlikely(priority < 0 || priority > PRIORITY_MAX))
 		return -EINVAL;
 
-	return encapsulate(frame->pkt, priority, &frame->gk_conf->net->back,
+	if (direct_if_possible != 0 && priority == PRIORITY_GRANTED)
+		return update_pkt_priority(frame->packet, priority);
```

Comment:
`encapsulate()` makes sure that there is enough space for the Ethernet header because the code assumes so later on. Since `update_pkt_priority()` is taking over the role of `encapsulate()` here, it needs to do the same.

To understand why we need to check that there's enough space for the Ethernet header, consider the example: front net doesn't use VLAN, but the back net does. So the Ethernet header at the back is larger. The reverse requires the Ethernet header to shrink.